### ISSUE TITLE
PROPOSED FIX: Changes needed for react-rails 2.2.0 to work properly

### ISSUE THREAD

**Issue by JohnAtFenestra** [suzan2go/react-rails-redux-sample#14]
The example fails with react-rails 2.2. It generates the following error:

```
ActionView::Template::Error (Encountered error "#<ExecJS::ProgramError: TypeError: Cannot read property 'serverRender' of undefined>" when prerendering Root with {"counter":10}
```

To get the sample to work with react-rails 2.2

1. From the CLI, execute the following

```
rails g react:install
```
When the routine asks about the conflict with `components.js`, ignore it. Keep the current file.

2. `actions/counter.js` must be broken up into two pieces:

move the constant declarations to a new file: `counterTypes.js`

```javascript
export const INCREMENT_COUNTER = 'INCREMENT_COUNTER';
export const DECREMENT_COUNTER = 'DECREMENT_COUNTER';
export const SET_COUNTER = 'SET_COUNTER';
```
add the following to `actions/counter.js`:

```javascript
import {SET_COUNTER, INCREMENT_COUNTER, DECREMENT_COUNTER} from './counterTypes';
```

add the following to `reducers/counter.js`:

```javascript
import { INCREMENT_COUNTER, DECREMENT_COUNTER, SET_COUNTER } from '../actions/counterTypes';
```

At this point the application should run properly. 

NOTE: this has been tested both with the current dependencies: Ruby 2.3.3 and Rails 5.0.0.1 and also Ruby 2.4 and Rails 5.1.

**Comment by suusan2go** [comment]
@JohnAtFenestra Thanks for your issue reporting 🍺  Please send me pull requests !

**Comment by suusan2go** [comment]
closed via #15
